### ISSUE TITLE
[bat] Building not working as expected

### ISSUE THREAD

**Issue by baehless** [netsec-ethz/scion-apps#18]
Following the instructions in the Readme leads to the following errors.
```
alessio@alessio-System-Product-Name:~/go/src/github.com/netsec-ethz/scion-apps/bat$ govendor sync
alessio@alessio-System-Product-Name:~/go/src/github.com/netsec-ethz/scion-apps/bat$ govendor add +e
alessio@alessio-System-Product-Name:~/go/src/github.com/netsec-ethz/scion-apps/bat$ go install
vendor/github.com/lucas-clemente/quic-go/h2quic/client.go:18:2: use of internal package not allowed
vendor/github.com/lucas-clemente/quic-go/h2quic/client.go:19:2: use of internal package not allowed
```

**Comment by chaehni** [comment]
Can you post a `tree -d` of your vendor folder?

**Comment by matzf** [comment]
The problem seems to be that `github.com/lucas-clemente/quic-go` is contained in more than one vendor directory); it was added by running `govender add +e` (as instructed by the readme).

My two cents on this are:
* we should avoid using multiple vendor folders at the same time; this means either
  1. have only a single vendor folder in the root of the project, XOR
  2. have (exactly) one vendor folder for each 
* we should remove the vendoring in lib/shttp; if we choose option 2, then move it to the lib/shttp/examples directory
* we should include scion (on the scionlab branch) and all required dependencies in the vendor.json and remove the `govendor add +e` step from the READMEs. In particular, we should configure it properly with the path `scion/scionproto` but with the *origin* `netsec-ethz/netsec-scion`.
Also, add instructions on how to update the vendored versions.

**Comment by baehless** [comment]
Tried again with a fresh copy of the scion-app repository and now I get something different.
```
alessio@alessio-System-Product-Name:~/go/src/github.com/netsec-ethz/scion-apps$ govendor sync
alessio@alessio-System-Product-Name:~/go/src/github.com/netsec-ethz/scion-apps$ cd bat/
alessio@alessio-System-Product-Name:~/go/src/github.com/netsec-ethz/scion-apps/bat$ govendor sync
alessio@alessio-System-Product-Name:~/go/src/github.com/netsec-ethz/scion-apps/bat$ govendor add +e
alessio@alessio-System-Product-Name:~/go/src/github.com/netsec-ethz/scion-apps/bat$ go install
# github.com/netsec-ethz/scion-apps/bat/vendor/github.com/lucas-clemente/quic-go/qerr
vendor/github.com/lucas-clemente/quic-go/qerr/quic_error.go:54:2: undefined: utils.Errorf
# github.com/netsec-ethz/scion-apps/bat/vendor/github.com/marten-seemann/qtls
vendor/github.com/marten-seemann/qtls/common.go:300:18: undefined: tls.ECDSAWithSHA1
```

And the vendor folder structure is the following.

```
├── github.com
│   ├── bifurcation
│   │   └── mint
│   │       └── syntax
│   ├── cheekybits
│   │   └── genny
│   │       └── generic
│   ├── dchest
│   │   └── cmac
│   ├── google
│   │   └── gopacket
│   │       └── layers
│   ├── go-stack
│   │   └── stack
│   ├── hashicorp
│   │   └── golang-lru
│   │       └── simplelru
│   ├── inconshreveable
│   │   └── log15
│   │       ├── ext
│   │       └── term
│   ├── kormat
│   │   └── fmt15
│   ├── lucas-clemente
│   │   ├── aes12
│   │   ├── quic-go
│   │   │   ├── h2quic
│   │   │   ├── internal
│   │   │   │   ├── ackhandler
│   │   │   │   ├── congestion
│   │   │   │   ├── crypto
│   │   │   │   ├── flowcontrol
│   │   │   │   ├── handshake
│   │   │   │   ├── protocol
│   │   │   │   ├── qerr
│   │   │   │   ├── utils
│   │   │   │   └── wire
│   │   │   └── qerr
│   │   └── quic-go-certificates
│   ├── marten-seemann
│   │   └── qtls
│   ├── mattn
│   │   ├── go-colorable
│   │   └── go-isatty
│   ├── netsec-ethz
│   │   └── scion-apps
│   │       └── lib
│   │           ├── scionutil
│   │           └── shttp
│   ├── patrickmn
│   │   └── go-cache
│   ├── pierrec
│   │   ├── lz4
│   │   └── xxHash
│   │       └── xxHash32
│   └── scionproto
│       └── scion
│           └── go
│               ├── lib
│               │   ├── addr
│               │   ├── assert
│               │   ├── common
│               │   ├── crypto
│               │   │   ├── cert
│               │   │   └── trc
│               │   ├── ctrl
│               │   │   ├── cert_mgmt
│               │   │   ├── path_mgmt
│               │   │   └── seg
│               │   ├── hpkt
│               │   ├── infra
│               │   │   ├── disp
│               │   │   └── transport
│               │   ├── l4
│               │   ├── log
│               │   ├── overlay
│               │   ├── pathmgr
│               │   ├── pktcls
│               │   ├── sciond
│               │   ├── scmp
│               │   ├── snet
│               │   │   └── squic
│               │   ├── sock
│               │   │   └── reliable
│               │   ├── spath
│               │   │   └── spathmeta
│               │   ├── spkt
│               │   ├── util
│               │   └── xtest
│               │       └── graph
│               └── proto
├── golang.org
│   └── x
│       ├── crypto
│       │   ├── chacha20poly1305
│       │   ├── curve25519
│       │   ├── ed25519
│       │   │   └── internal
│       │   │       └── edwards25519
│       │   ├── hkdf
│       │   ├── internal
│       │   │   ├── chacha20
│       │   │   └── subtle
│       │   └── poly1305
│       ├── net
│       │   ├── context
│       │   ├── http
│       │   │   └── httpguts
│       │   ├── http2
│       │   │   └── hpack
│       │   ├── idna
│       │   └── lex
│       │       └── httplex
│       ├── sys
│       │   ├── cpu
│       │   └── unix
│       └── text
│           ├── collate
│           │   └── build
│           ├── internal
│           │   ├── colltab
│           │   ├── gen
│           │   ├── language
│           │   │   └── compact
│           │   ├── tag
│           │   ├── triegen
│           │   └── ucd
│           ├── language
│           ├── secure
│           │   └── bidirule
│           ├── transform
│           └── unicode
│               ├── bidi
│               ├── cldr
│               ├── norm
│               └── rangetable
├── gopkg.in
│   ├── natefinch
│   │   └── lumberjack.v2
│   ├── restruct.v1
│   └── yaml.v2
└── zombiezen.com
    └── go
        └── capnproto2
            ├── encoding
            │   └── text
            ├── internal
            │   ├── nodemap
            │   ├── packed
            │   ├── schema
            │   └── strquote
            ├── pogs
            └── schemas
```

**Comment by chaehni** [comment]
@baehless Are you using the SCION version pointed to by netsec-ethz/scionlab?

@matzf The idea was not to download what is on the system anyways and, more importantly, use whatever version of SCION is installed to ensure compatibility. However, this is causing too many problems now. I agree on the two options and I'd prefer option 1.

**Comment by baehless** [comment]
Yes, I am.
```
alessio@alessio-System-Product-Name:~/go/src/github.com/scionproto/scion$ git remote -v
origin	https://github.com/netsec-ethz/netsec-scion (fetch)
origin	https://github.com/netsec-ethz/netsec-scion (push)
```

**Comment by chaehni** [comment]
@baehless This is weird. The vendor.json file pins the versions of all dependencies. The user should never be required to up-/downgrade any dependency manually.

**Comment by chaehni** [comment]
fixed by 0bf02f2449eab45198ad5ce6c46821b30b09b095
